src/app/containers/InlineLink/index.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { string, node } from 'prop-types';
+import InlineLink from '../../components/InlineLink';
+
+const InternalInlineLink = InlineLink.withComponent(Link);
+
+const InlineLinkContainer = ({ href, children, ...rest }) => {
+  // if URL matches a valid route, use a react-router link
+  if (href.match('^/news/articles/c[a-zA-Z0-9]{10}o$')) {
```

Comment:
You should move this regex out to a constant in a library file, since we use it in this file too `/containers/Article/index.jsx` and will likely need to re-use it in other places as well.

Reply:
Also this check `is a valid article route` should be a library function that we import and use in this file.

Reply:
Would you accept doing all this in a follow-up PR?